packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts

```diff
@@ -468,13 +467,13 @@ export class AdapterDayjs implements MuiPickersAdapter<Dayjs> {
     return nestedWeeks;
   };
 
-  public getWeekNumber = (date: Dayjs) => {
-    return date.week();
+  public getWeekNumber = (value: Dayjs) => {
+    return value.week();
   };
 
   public getYearRange = (start: Dayjs, end: Dayjs) => {
-    const startDate = this.dayjs(start).startOf('year');
```

Comment:
No need to re-parse here (same on `getWeekArray`), `startOf` and `endOf` are not mutating the initial date